File: app/src/main/java/com/manimarank/spell4wiki/data/auth/WikiAuthenticatorService.java
package com.manimarank.spell4wiki.data.auth;

import android.accounts.AbstractAccountAuthenticator;
import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

import androidx.annotation.Nullable;

public class WikiAuthenticatorService extends Service {

    @Nullable private AbstractAccountAuthenticator authenticator;

    @Override
    public void onCreate() {
        super.onCreate();
        authenticator = new WikiAuthenticator(this);
    }
    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return authenticator == null ? null : authenticator.getIBinder();
    }
}

File: app/src/main/java/com/manimarank/spell4wiki/data/auth/WikiAuthenticatorService.kt
package com.manimarank.spell4wiki.data.auth

import android.accounts.AbstractAccountAuthenticator
import android.app.Service
import android.content.Intent
import android.os.IBinder

class WikiAuthenticatorService : Service() {
    private var authenticator: AbstractAccountAuthenticator? = null
    override fun onCreate() {
        super.onCreate()
        authenticator = WikiAuthenticator(this)
    }

    override fun onBind(intent: Intent): IBinder? {
        return authenticator?.iBinder
    }
}